pkg/ingester/wal.go

```diff
@@ -180,11 +181,14 @@ func (w *walWrapper) run() {
 
 const checkpointPrefix = "checkpoint."
 
-func (w *walWrapper) performCheckpoint() (err error) {
+func (w *walWrapper) performCheckpoint(immediate bool) (err error) {
 	if !w.cfg.CheckpointEnabled {
 		return nil
 	}
 
+	w.checkpointMtx.Lock()
```

Comment:
Please add a comment to explain why we do it. May not be much clear why we lock the entire function.